src/Core/Domain/AuthorizationServer/Command/EditApplicationCommand.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\Exception\ApplicationConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\ValueObject\ApplicationId;
+use PrestaShop\PrestaShop\Core\Domain\AuthorizationServer\ValueObject\ApplicationIdInterface;
+
+/**
+ * Edit application with provided data
+ */
+class EditApplicationCommand
+{
+    /**
+     * @var ApplicationIdInterface
+     */
+    private $applicationId;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string|null
+     */
+    private $description;
+
+    /**
+     * @param int $applicationId
+     *
+     * @throws ApplicationConstraintException
+     */
+    public function __construct(int $applicationId)
+    {
+        $this->applicationId = new ApplicationId($applicationId);
+    }
+
+    /**
+     * @return ApplicationIdInterface
+     */
+    public function getApplicationId(): ApplicationIdInterface
+    {
+        return $this->applicationId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return self
+     */
+    public function setName(string $name): self
```

Comment:
In general, commands should be immutable. See https://cqrs.nu/Faq/commands-and-events#:~:text=Commands%20are%20immutable%20because%20their,took%20place%20in%20the%20past.